### ISSUE TITLE
UI: Button rollover does not switch back to idle state

### ISSUE THREAD

**Issue by reneandritsch** [RxLaboratory/DuAEF_Duik#361]
# Environement
### Tool & Version
- Duik 16.1.0-Alpha-4

### After Effects
- CC2018 (15.1.2)
- English

### OS
- Mac OS 10.13.6

# What's the issue?
- Bug Report

### Steps to reproduce

1. Move the mouse over the button 
2. Move mouse away from button to a non-button area

### Expected behaviour
Buttons should become red and then white again if mouse cursor leaves button

### Actual behaviour
Button stays red if not triggered by a different button.

![ui_roll-over](https://user-images.githubusercontent.com/38455050/48677688-d8c21100-eb78-11e8-96e1-6808522f94be.gif)

**Comment by Nico-Duduf** [comment]
This is because of some limitations in the tools in Ae available to build the user interface, but I may have a workaround I'll have to test...

try:
replace buttons margins by statictext placeholder and when over these placeholders, dim corresponding button.

**Comment by reneandritsch** [comment]
Thanks for your efforts! Have a good start into the week! René

**Comment by Nico-Duduf** [comment]
I did not manage to fix this, there are too many weird things hapenning when I try to hack how Ae manages the display ^^
I'll try again for another future version, maybe I'll have new ideas...

**Comment by reneandritsch** [comment]
Dear Nicolas, thank you for trying!

**Comment by LarsJandel** [comment]
Hi Duduf, just saw this entry by coincidence...

My guess: You are just missing `eventListener` elements to fill the empty space, which would call the dim function. Elements below groups/buttons aligned to `top`, to the sides of `center` groups/buttons or another example, between the notepad icon to the `left` and the tools to the `right`. Having `eventListener` elements which are set to `['fill','fill']` the empty space should work to call the dim function? Maybe this is just what you already tried when you mentioned swapping margins for placeholders...   😉

Cheers, Lars.

**Comment by Nico-Duduf** [comment]
Hi Lars!

I tried this, but I had two issues: the event is not triggered if the mouse moves too fast (which makes sense as the coordinates would "jump" above the placeholder between two iterations of the event loop)
And, weirder, the empty statictext set to `fill` does not actually fill the whole gap, there is some empty space, a spacing between my actual button and these fake margins which should not be there. This does not happen if I had a character in the placeholder, which could be something invisible, but I don't know what (as it may have to not exceed 3 pixels width).

In other words: I'm really tired about ScriptUI and should move to CEP ;) Just need to find some time to do this, it will be quite a lot of work for Duik ; I'd have to create another smaller extension just to test and learn first

**Comment by LarsJandel** [comment]
Hrhr, so true. CEP - yes...
Yep, cursor speed is an issue - people shouldn't be that fast. 😁
I always use invisible buttons - maybe they work better than text?
`btn_placeholder.onDraw = function(){};`

**Comment by Nico-Duduf** [comment]
Wait... That's an great idea these invisible buttons!
I'll try this, thanks!

Le 03/03/2019 à 11:06, LarsJandel a écrit :

**Comment by reneandritsch** [comment]
That sounds great!
